src/hdf5plugin/_filters.py

```diff
@@ -482,6 +486,112 @@ def __init__(self,
         logger.info(f"filter options = {self.filter_options}")
 
 
+class Sperr(_FilterRefClass):
+    """``h5py.Group.create_dataset``'s compression arguments for using SPERR filter.
+
+    It can be passed as keyword arguments:
+
+    .. code-block:: python
+
+        f = h5py.File('test.h5', 'w')
+        f.create_dataset(
+            'sperr',
+            data=numpy.random.random(1000).reshape(100, 10),
+            **hdf5plugin.Sperr(rate=16))
+        f.close()
+
+    This filter provides different modes:
+
+    - **Fixed bit-per-pixel (BPP)**:
+      The quality argument provides the target bitrate (range: 0.0 < quality < 64.0)
+
+      .. code-block:: python
+
+        f.create_dataset(
+            'sperr_fixed_bit-per-pixel',
+            data=numpy.random.random(1000).reshape(100, 10),
+            **hdf5plugin.Sperr(rate=10))
+
+    - **Fixed peak signal-to-noise ratio (PSNR)**:
+      The quality argument provides the target PSNR (range: 0.0 < quality)
+
+      .. code-block:: python
+
+        f.create_dataset(
+            'sperr_fixed_peak_signal-to-noise_ratio',
+            data=numpy.random.random(1000).reshape(100, 10),
+            **hdf5plugin.Sperr(peak_signal_to_noise_ratio=1e-6))
+
+    - **Fixed point-wise error (PWE)**:
+      The quality argument provides the PWE tolerance (range: 0.0 < quality)
+
+      .. code-block:: python
+
+        f.create_dataset(
+            'sperr_fixed_point-wise_error',
+            data=numpy.random.random(1000).reshape(100, 10),
+            **hdf5plugin.Sperr(point_wise_error=1e-4))
+
+    For more details, see `H5Z-SPERR <https://github.com/NCAR/H5Z-SPERR>`_.
+    """
+    filter_name = "sperr"
+    filter_id = SPERR_ID
+
+    _FRACTIONAL_BITS = 16
+    _INTEGER_BITS = 12
+
+    def __init__(
+            self,
+            rate: float | None = None,
+            peak_signal_to_noise_ratio: float | None = None,
+            point_wise_error: float | None = None):
```

Comment:
Instead of passing `mode` and `quality` as in the C filter, I choose to expose one argument for each mode as it is already done for `ZFP`, `SZ` and `SZ3` and tried to use similar names.

@shaomeng, from what I understood "point-wise error" tolerance is an absolute tolerance, am I right?
In this case I would rename it to `absolute` (as `SZ` and `SZ3`) or `accuracy` (as `Zfp`) (and it would have been best if we avoided this difference...)

Reply:
It's correct that "point-wise error" is equivalent to "absolute error" so I'm happy to see it using that name too.

Reply:
Then I'll use `absolute` as keyword argument